pkg/gatewayserver/io/io.go

```diff
@@ -99,6 +100,15 @@ type Connection struct {
 
 	statsChangedCh chan struct{}
 	locCh          chan struct{}
+
+	lastUp            *uplinkMessage
+	lastRepeatUpEvent time.Time
+}
+
+type uplinkMessage struct {
+	payload     []byte
+	frequency   uint64
+	antennaMask uint64
```

Comment:
Why a mask?

Reply:
This is to reduce memory allocations per uplink message.

Is it safe to assume that antenna index will be < 63, or is this a prime example of premature (and largely unwarranted) optimization?

Reply:
Although in practice this is most likely safe to assume, there is nothing to guarantee it, so it's not ideal. A `map[uint32]struct{}` has a bit of a memory footprint, but could work.

If you want to keep this mask solution, at least document it (especially the assumptions).

Reply:
Well, if we are gonna do `map[uint32]struct{}`, we must as well just do the `[]uint32` instead. Better safe than sorry.